### PR TITLE
DM-6613: Typo in afw/display/interface.py

### PR DESCRIPTION
**Pull request by fred3m**
_callbacks property was missing an underscore.

### CODE REVIEW

File: python/lsst/afw/display/interface.py [hunk 1]
```diff
@@ -523,27 +526,26 @@ def pan(self, colc=None, rowc=None, origin=afwImage.PARENT):
...
-            except Exception, e:
-                print >> sys.stderr, "Display.callbacks[%s](%s, %s, %s) failed: %s" % \
-                    (k, k, x, y, e)
+            
+            if k not in self._callbacks:
+                logger.warn("No callback registered for {0}".format(k))
+            else:
+                try:
+                    interact_finished = self._callbacks[k](k, x, y)
+                except Exception, e:
```
Comment (timj) [inline python/lsst/afw/display/interface.py]: If these lines are changes can you please upgrade them to modern Exceptions?
```python
except Exception as e:
```

File: tests/display.py [hunk 2]
```diff
@@ -159,6 +159,13 @@ def testImageTypes(self):
 
             im = afwImage.MaskU(self.fileName, 3)
             dummy.mtv(im)
+    
+    def testInteract(self, timeout=1):
+        """Check that interact exists if a q, \c CR, or \c ESC is pressed.
+        If this is run using the virtualDevice a "q" is automatically triggered.
```
Comment (jdswinbank) [inline tests/display.py]: "exists" or "exits"?

File: tests/display.py [hunk 3]
```diff
@@ -159,6 +159,13 @@ def testImageTypes(self):
 
             im = afwImage.MaskU(self.fileName, 3)
             dummy.mtv(im)
+    
+    def testInteract(self, timeout=1):
```
Comment (jdswinbank) [inline tests/display.py]: `timeout` seems to be unused.

File: python/lsst/afw/display/interface.py [hunk 4]
```diff
@@ -523,27 +526,26 @@ def pan(self, colc=None, rowc=None, origin=afwImage.PARENT):
...
     def interact(self):
         """!Enter an interactive loop, listening for key presses in display and firing callbacks.
-
-        Exit with q, \c CR, or \c ESC
-    """
-
-        while True:
+            Exit with q, \c CR, or \c ESC
+        """
+        interact_finished = False
```
Comment (jdswinbank) [inline python/lsst/afw/display/interface.py]: [Coding standard](https://developer.lsst.io/coding/python_style_guide.html#naming-conventions) prefers `interactFinished`, ugly though it is.

File: python/lsst/afw/display/interface.py [hunk 5]
```diff
@@ -523,27 +526,26 @@ def pan(self, colc=None, rowc=None, origin=afwImage.PARENT):
...
-
+    
     def interact(self):
         """!Enter an interactive loop, listening for key presses in display and firing callbacks.
-
-        Exit with q, \c CR, or \c ESC
-    """
-
-        while True:
+            Exit with q, \c CR, or \c ESC
```
Comment (jdswinbank) [inline python/lsst/afw/display/interface.py]: In fact, we exit when any callback returns `True`, right? (I realise this isn't a new addition in your changes, though).

File: python/lsst/afw/display/interface.py [hunk 6]
```diff
@@ -523,27 +526,26 @@ def pan(self, colc=None, rowc=None, origin=afwImage.PARENT):
         """!Pan to (rowc, colc); see also zoom"""
 
         self.zoom(None, colc, rowc, origin)
-
+
```
Comment (jdswinbank) [inline python/lsst/afw/display/interface.py]: Please don't add unnecessary whitespace.

File: python/lsst/afw/display/interface.py [hunk 7]
```diff
@@ -29,6 +29,9 @@
 import importlib
 import lsst.afw.geom  as afwGeom
 import lsst.afw.image as afwImage
+import lsst.log
+
+logger = lsst.log.Log.getLogger("afw.display.interact")
```
Comment (jdswinbank) [inline python/lsst/afw/display/interface.py]: I actually don't know the answer to this: do we have a convention for choosing logger names?

Regardless of a convention, the above looks a bit suspicious because it's in scope for the whole file but calls out `interact`, which is one specific method in one class contained in the file, by name.

File: tests/display.py [hunk 8]
```diff
@@ -159,6 +159,13 @@ def testImageTypes(self):
...
+        """Check that interact exists if a q, \c CR, or \c ESC is pressed.
```
Comment (jdswinbank) [inline tests/display.py]: "exists" or "exits"?

File: python/lsst/afw/display/interface.py (hunk 7 above)
Comment (hchiang2) [inline python/lsst/afw/display/interface.py]: While doing the logging framework transition I have the exactly same question about the convention of logger names. If the log belongs to a Task the logger name is its `_DefaultName`.  If it's not a Task, the name seems a bit arbitrary and depends on the author's intent from I've seen so far.

File: python/lsst/afw/display/interface.py [hunk 9]
```diff
@@ -523,27 +526,26 @@ def pan(self, colc=None, rowc=None, origin=afwImage.PARENT):
...
-                print >> sys.stderr, "Display.callbacks[%s](%s, %s, %s) failed: %s" % \
-                    (k, k, x, y, e)
+            
+            if k not in self._callbacks:
+                logger.warn("No callback registered for {0}".format(k))
+            else:
+                try:
+                    interact_finished = self._callbacks[k](k, x, y)
+                except Exception, e:
+                    logger.error("Display._callbacks[{0}]({0},{1},{2}) failed: {3}".format(k,x,y,e))
```
Comment (jdswinbank) [inline python/lsst/afw/display/interface.py]: Need spaces after commas in the argument list — `k, x, y, e`.

File: python/lsst/afw/display/interface.py (hunk 7 above)
Comment (fred3m) [inline python/lsst/afw/display/interface.py]: This is a typo. It should have been 
`logger = lsst.log.Log.getogger("afw.display.interface")`.

If lsst.log is like the logging module, this can matter a great deal. For example, if the "afw" logger is set to a given level (logging.INFO for example), then "afw.display.interface" will inherit that setting. For this reason other projects that I've been involved in use this convention, although I have also seen it become more specific ("afw.display.interface.interact") in a function if there is a reason that the function might want to be singled out explicitly. I don't know enough about the C++ logger and what effect this might have, but if it has similar functionality I would recommend that we adopt a similar convention. Even if it isn't, it might still be useful for the python side of the code.

File: python/lsst/afw/display/interface.py (hunk 6 above)
Comment (fred3m) [inline python/lsst/afw/display/interface.py]: Sorry I didn't know that this is the convention. My text editor, Textmate, adds whitespace by default that matches the current indentation level (for the class in this case). Personally I like this style *much* better but I'll be more careful in the future.

Comment (timj) [inline python/lsst/afw/display/interface.py]: Yes, we try to remove trailing whitespace. We are also actively updating the style guide to be able to express it in terms of PEP-8.

File: tests/display.py (hunk 3 above)
Comment (fred3m) [inline tests/display.py]: This was corrected in a later commit. I rebased before pushing my commits but it didn't seem to fix this (but the current file [https://github.com/lsst/afw/blob/master/python/lsst/afw/display/interface.py](url) shows timeout removed).
